commons/zenoh-shm/src/api/provider/shm_provider.rs

```diff
@@ -182,8 +183,14 @@ where
         // NOTE: Depending on internal implementation, provider's backend might relayout
         // the allocations for bigger alignment (ex. 4-byte aligned allocation to 8-bytes aligned)
 
+        // Obtain nonzero size
+        let nonzero_size = data
```

Comment:
I believe this block can be removed since the conversion is now done inside `MemoryLayout::new` which will return `ZLayoutError::IncorrectLayoutArgs` in case of *zero size*.